rust/lance/src/dataset/optimize.rs

```diff
@@ -796,6 +816,7 @@ async fn rewrite_files(
         max_rows_per_file: options.target_rows_per_fragment,
         max_rows_per_group: options.max_rows_per_group,
         mode: WriteMode::Append,
+        object_store_registry,
         ..Default::default()
     };
     let mut new_fragments = write_fragments_internal(
```

Comment:
We explicitly pass a resolved object store below, so I don't think we need to pass down the registry here. I think the modifications to this file aren't necessary.

Reply:
yup, I think you're right; reverted.